### DISCUSSION TITLE
Does anyone know what the elt buffer size limit is?

### DISCUSSION THREAD

**Discussion by bikerdan** [meltano/meltano#7225] [Q&A]
What is the largest the elt.buffer_size can be set to?  Would it be possible to set it to say 2GB?

**Comment by tayloramurphy** [comment]
@aaronsteers @edgarrmondragon do you all know?

**Reply by edgarrmondragon** [reply]
Not off the top of my head, but the absolute upper limit and how to query it surely changes from system to system. Safe values are probably something well below the memory constraints of the computer or container.

**Reply by bikerdan** [reply]
So as far as we know, the limits are with the system itself?  Or could there be limits on the code and libraries used for the buffer?

**Reply by edgarrmondragon** [reply]
Yeah, the limits are in the system itself. The value is passed to [`asyncio.create_subprocess_exec`](https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.create_subprocess_exec), and there are no hard limits there afaict.

**Reply by aaronsteers** [reply]
Agreed with @edgarrmondragon above. My understanding is that this is only limited by the available RAM on the machine.

Larger values will increase the amount of unprocessed data that can be held in the target's input buffer before the tap must be paused. If there's enough RAM to hold the entire dataset in memory, then in theory the tap could run entirely to completion without ever needing to be paused. (Although the target's own performance and runtime duration may still be unchanged.)

**Reply by bikerdan** [reply]
Thanks for the clarification.
